SerialPrograms/Source/Integrations/DiscordSocial/DiscordSocial.cpp

```diff
@@ -0,0 +1,171 @@
+#ifdef PA_SOCIAL_SDK
+#define DISCORDPP_IMPLEMENTATION
+
+#include <thread>
+#include "DiscordSocial.h"
+#include "CommonFramework/Logging/Logger.h"
+#include "Integrations/ProgramTracker.h"
+#include "../../CommonFramework/GlobalSettingsPanel.h"
+#include "../../CommonFramework/Globals.h"
+
+using namespace discordpp;
+namespace PokemonAutomation{
+namespace Integration{
+namespace DiscordSocialSDK{
+
+DiscordSocial& DiscordSocial::instance(){
+    static DiscordSocial instance;
+    return instance;
+}
+
+void DiscordSocial::run(){
+    m_client = std::make_shared<Client>();
+    if (!m_client){
+        log("Failed to initialize DiscordSocialSDK.", "run()", LoggingSeverity::Error);
+        m_client.reset();
+        return;
+    }
+
+    m_client->SetApplicationId(m_app_id);
+    m_client->AddLogCallback([&](auto message, auto severity){
+        log(message, "Internal", severity);
+    }, m_log_level);
+
+    std::thread(&DiscordSocial::thread_loop, this).detach();

```

Comment:
Never use detach since it's impossible to clean up due to having an unbounded lifetime. If you destruct this class, and the thread is still running, you dangle all the fields that it references.

Store the thread into the class. Then in the destructor after you signaled the thread to exit, join the thread.
```
m_thread = std::thread(&DiscordSocial::thread_loop, this);
```

In destructor:
```
m_running.store(false, std::memory_order_release);
m_thread.join();

//  Now it's safe to destroy the client.
if (m_client) m_client.reset();
```